### PR TITLE
[Refactor] refactor BaseTableInfo to make it better

### PR DESCRIPTION
**Pull request by ABingHuang**
## Why I'm doing:
Now BaseTableInfo class was previously too complex and coupled with too much logic, and the usage of it is error prone.

## What I'm doing:
1. make BaseTableInfo simple, just store properties without logic
2. add more check to avoid npe
3. make it more reusable

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by satanson — DISMISSED** [review]

**Review by satanson — DISMISSED** [review]

**Review by LiShuMing — DISMISSED** [review]

File: fe/fe-core/src/test/java/com/starrocks/catalog/BaseTableInfoTest.java [hunk 1]
```diff
@@ -1,50 +0,0 @@
...
-package com.starrocks.catalog;
-
-import com.starrocks.server.CatalogMgr;
-import com.starrocks.server.GlobalStateMgr;
-import mockit.Expectations;
-import mockit.Mocked;
-import org.junit.Assert;
-import org.junit.Test;
-
-public class BaseTableInfoTest {
```
Comment (LiShuMing) [inline fe/fe-core/src/test/java/com/starrocks/catalog/BaseTableInfoTest.java:25]: Why remove this?

Comment (ABingHuang) [reply fe/fe-core/src/test/java/com/starrocks/catalog/BaseTableInfoTest.java:25]: BaseTableInfo becomes a POJO class. The tests are useless

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 2]
```diff
@@ -922,7 +924,11 @@ private void replaceTableAlias(SlotRef slotRef,
             List<BaseTableInfo> baseTableInfos = statement.getBaseTableInfos();
             for (BaseTableInfo baseTableInfo : baseTableInfos) {
                 if (table.isNativeTableOrMaterializedView()) {
-                    if (baseTableInfo.getTable().equals(table)) {
+                    Optional<Table> tableOptional = MvUtils.getTableWithIdentifier(baseTableInfo);
+                    if (!tableOptional.isEmpty()) {
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:928]:
```suggestion
                    if (tableOptional.isEmpty()) {
```
should be that?

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java]: yes

**Review by Seaven — APPROVED** [review]

**Review by LiShuMing — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**